src/gtfs/mod.rs

```diff
@@ -341,6 +348,201 @@ pub struct Configuration {
+    route_id: Option<String>,
+    trip_id: Option<String>,
+    #[serde(deserialize_with = "de_opt_bool_from_str")]
+    is_operator: Option<bool>,
+    organization_name: String,
+    attribution_url: Option<String>,
+    attribution_email: Option<String>,
+    attribution_phone: Option<String>,
+}
+
+impl Hash for Attribution {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.organization_name.hash(state);
+        self.attribution_url.hash(state);
+        self.attribution_email.hash(state);
+        self.attribution_phone.hash(state);
+    }
+}
+
+#[derive(Eq, Hash, PartialEq)]
+struct AttributionRule {
+    id: String,
+    object_type: ObjectType,
+    object_id: String,
+    organization_name: String,
+    attribution_url: Option<String>,
+    attribution_email: Option<String>,
+    attribution_phone: Option<String>,
+}
+
+impl TryFrom<&Attribution> for AttributionRule {
+    type Error = Error;
+    fn try_from(attribution: &Attribution) -> Result<Self> {
+        let (object_type, object_id) = match (
+            attribution.route_id.is_some(),
+            attribution.trip_id.is_some(),
+        ) {
+            (true, false) => (Some(ObjectType::Route), attribution.route_id.clone()),
+            (false, true) => (
+                Some(ObjectType::VehicleJourney),
+                attribution.trip_id.clone(),
+            ),
+            _ => (None, None),
+        };
+
+        if object_type.is_none() {
+            let message = if let Some(attribution_id) = &attribution.attribution_id {
+                format!(
+                    "Attribution {} must have either route_id or trip_id",
+                    attribution_id
+                )
+            } else {
+                format!(
+                    "Attribution {:?} must have either route_id or trip_id",
+                    attribution
+                )
+            };
+            bail!(message);
+        }
+
+        let attribution_rule = AttributionRule {
+            id: calculate_hash(&attribution).to_string(),
+            object_type: object_type.expect("an error occured"),
+            object_id: object_id.expect("an error occured"),
+            organization_name: attribution.organization_name.clone(),
+            attribution_url: attribution.attribution_url.clone(),
+            attribution_email: attribution.attribution_email.clone(),
+            attribution_phone: attribution.attribution_phone.clone(),
+        };
+        Ok(attribution_rule)
+    }
+}
+
+fn read_attributions<H>(file_handler: &mut H, file_name: &str) -> Result<Vec<AttributionRule>>
+where
+    for<'a> &'a mut H: FileHandler,
+{
+    let (reader, path) = file_handler.get_file_if_exists(file_name)?;
+    let file_name = path.file_name();
+    let basename = file_name.map_or(path.to_string_lossy(), |b| b.to_string_lossy());
+    let mut attribution_rules = Vec::new();
+    if let Some(reader) = reader {
+        info!("Reading {}", basename);
```

Comment:
Structured log can be used like this.

```suggestion
        info!(file_name = %basename, "Reading file");
```

If you ever want to parse the logs, or filter the logs based on some value, doing it like this will help.